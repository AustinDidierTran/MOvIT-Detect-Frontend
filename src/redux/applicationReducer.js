// ------------------------------------
// Constants
// ------------------------------------

let url;

switch (process.env.NODE_ENV) {
  case 'production':
  case 'pi':
    url = 'http://192.168.4.1:1880/';
    break;
  case 'local':
    url = 'http://localhost:1880/';
    break;
  default:
    url = 'https://private-f2484-movitplus.apiary-mock.com/';
}

export const URL = url;
export const LANGUAGE = 'LANGUAGE';
export const FR = 'FR';
export const EN = 'EN';
export const PROFILE = 'PROFILE';
export const TOKEN = 'TOKEN';

// ------------------------------------
// Actions
// ------------------------------------
function changeLanguage() {
  return {
    type: LANGUAGE,
  };
}

function changeProfile(profileName) {
  return {
    type: PROFILE,
    profile: profileName,
  };
}

function changeToken(tokenString) {
  return {
    type: TOKEN,
    token: tokenString,
  };
}

export const ApplicationActions = {
  changeLanguage,
  changeProfile,
  changeToken,
};
// ------------------------------------
// Action Handlers
// ------------------------------------
const ACTION_HANDLERS = {
  [LANGUAGE]: state => (
    { ...state, language: state.language === FR ? EN : FR }
  ),
  [PROFILE]: (state, action) => (
    { ...state, profile: action.profile }
  ),
  [TOKEN]: (state, action) => (
    { ...state, token: action.token, header: { headers: { Authorization: action.token } } }
  ),
};

// ------------------------------------
// Reducer
// ------------------------------------

export const initApplication = {
  language: 'FR',
  profile: '',
  userName: '',
  userID: '',
  token: null,
  header: {},
  maxAngle: null,
  userWeight: null,
};
export default function applicationReducer(state = initApplication, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
