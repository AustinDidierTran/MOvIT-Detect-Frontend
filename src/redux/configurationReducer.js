export const DATA_AGREEMENT = 'DATA_AGREEMENT';
export const LIGHT_AGREEMENT = 'LIGHT_AGREEMENT';
export const NOTIFICATION_AGREEMENT = 'NOTIFICATION_AGREEMENT';
export const DATA_DISAGREE_PERDIOD = 'DATA_DISAGREE_PERDIOD';
export const LIGHT_DISAGREE_PERDIOD = 'LIGHT_DISAGREE_PERDIOD';
export const NOTIFICATION_DISAGREE_PERDIOD = 'NOTIFICATION_DISAGREE_PERDIOD';

// -------------- CHECKBOX --------------
function changeDataAgreement() {
  return {
    type: DATA_AGREEMENT,
  };
}
function changeLightAgreement() {
  return {
    type: LIGHT_AGREEMENT,
  };
}
function changeNotificationAgreement() {
  return {
    type: NOTIFICATION_AGREEMENT,
  };
}

// -------------- TEXT --------------
function changeDataDisagreePeriod(period) {
  return {
    type: DATA_DISAGREE_PERDIOD,
    dataDisagreePeriod: period,
  };
}
function changeLightDisagreePeriod(period) {
  return {
    type: LIGHT_DISAGREE_PERDIOD,
    lightDisagreePeriod: period,
  };
}
function changeNotificationDisagreePeriod(period) {
  return {
    type: NOTIFICATION_DISAGREE_PERDIOD,
    notificationDisagreePeriod: period,
  };
}

export const ConfigurationActions = {
  changeDataAgreement,
  changeLightAgreement,
  changeNotificationAgreement,
  changeDataDisagreePeriod,
  changeLightDisagreePeriod,
  changeNotificationDisagreePeriod,
};

const ACTION_HANDLERS = {
  [DATA_AGREEMENT]: state => (
    { ...state, dataAgreement: !state.dataAgreement }
  ),
  [LIGHT_AGREEMENT]: state => (
    { ...state, lightAgreement: !state.lightAgreement }
  ),
  [NOTIFICATION_AGREEMENT]: state => (
    { ...state, notificationAgreement: !state.notificationAgreement }
  ),
  [DATA_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, dataDisagreePeriod: action.dataDisagreePeriod }
  ),
  [LIGHT_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, lightDisagreePeriod: action.lightDisagreePeriod }
  ),
  [NOTIFICATION_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, notificationDisagreePeriod: action.notificationDisagreePeriod }
  ),
};

export const initConfiguration = {
  dataAgreement: true,
  lightAgreement: true,
  notificationAgreement: true,
  dataDisagreePeriod: null,
  lightDisagreePeriod: null,
  notificationDisagreePeriod: null,
};

export default function applicationReducer(state = initConfiguration, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
