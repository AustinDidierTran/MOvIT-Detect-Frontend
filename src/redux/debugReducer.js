/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

export const IS_LED_BLINKING_ENABLED = 'IS_LED_BLINKING_ENABLED';
export const IS_VIBRATION_ENABLED = 'IS_VIBRATION_ENABLED';

// -------------- CHECKBOX --------------
function changeIsLedBlinkingEnabled(isEnabled) {
  return {
    type: IS_LED_BLINKING_ENABLED,
    isLedBlinkingEnabled: isEnabled,
  };
}
function changeIsVibrationEnabled(isEnabled) {
  return {
    type: IS_VIBRATION_ENABLED,
    isVibrationEnabled: isEnabled,
  };
}

export const DebugActions = {
  changeIsLedBlinkingEnabled,
  changeIsVibrationEnabled,
};

const ACTION_HANDLERS = {
  [IS_LED_BLINKING_ENABLED]: (state, action) => (
    { ...state, isLedBlinkingEnabled: action.isLedBlinkingEnabled }
  ),
  [IS_VIBRATION_ENABLED]: (state, action) => (
    { ...state, isVibrationEnabled: action.isVibrationEnabled }
  ),
};

export const initParameter = {
  isLedBlinkingEnabled: true,
  isVibrationEnabled: true,
};

export default function applicationReducer(state = initParameter, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
