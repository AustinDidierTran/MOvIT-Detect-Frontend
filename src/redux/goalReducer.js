/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

export const TILT_FREQUENCY_GOAL = 'TILT_FREQUENCY_GOAL';
export const TILT_LENGTH_GOAL = 'TILT_LENGTH_GOAL';
export const TILT_ANGLE_GOAL = 'TILT_ANGLE_GOAL';


// -------------- SLIDERS --------------

function changeTiltFrequencyGoal(tiltFreq) {
  return {
    type: TILT_FREQUENCY_GOAL,
    tiltFrequencyGoal: tiltFreq,
  };
}
function changeTiltLengthGoal(tiltLen) {
  return {
    type: TILT_LENGTH_GOAL,
    tiltLengthGoal: tiltLen,
  };
}
function changeTiltAngleGoal(tiltAng) {
  return {
    type: TILT_ANGLE_GOAL,
    tiltAngleGoal: tiltAng,
  };
}

export const GoalActions = {
  changeTiltFrequencyGoal,
  changeTiltLengthGoal,
  changeTiltAngleGoal,
};

const ACTION_HANDLERS = {
  [TILT_FREQUENCY_GOAL]: (state, action) => (
    { ...state, tiltFrequencyGoal: action.tiltFrequencyGoal }
  ),
  [TILT_LENGTH_GOAL]: (state, action) => (
    { ...state, tiltLengthGoal: action.tiltLengthGoal }
  ),
  [TILT_ANGLE_GOAL]: (state, action) => (
    { ...state, tiltAngleGoal: action.tiltAngleGoal }
  ),
};

// ------------------------------------
// Reducer
// ------------------------------------

export const initGoal = {
  tiltFrequencyGoal: 0,
  tiltLengthGoal: 0,
  tiltAngleGoal: 0,
};

export default function applicationReducer(state = initGoal, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}

