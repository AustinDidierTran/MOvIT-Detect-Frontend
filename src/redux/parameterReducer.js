/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

export const DATA_AGREEMENT = 'DATA_AGREEMENT';
export const LIGHT_AGREEMENT = 'LIGHT_AGREEMENT';
export const NOTIFICATION_AGREEMENT = 'NOTIFICATION_AGREEMENT';
export const DATA_DISAGREE_PERIOD = 'DATA_DISAGREE_PERDIOD';
export const LIGHT_DISAGREE_PERIOD = 'LIGHT_DISAGREE_PERDIOD';
export const NOTIFICATION_DISAGREE_PERDIOD = 'NOTIFICATION_DISAGREE_PERDIOD';

// -------------- CHECKBOX --------------
function changeDataAgreement(agree) {
  return {
    type: DATA_AGREEMENT,
    dataAgreement: agree,
  };
}
function changeLightAgreement(agree) {
  return {
    type: LIGHT_AGREEMENT,
    lightAgreement: agree,
  };
}
function changeNotificationAgreement(agree) {
  return {
    type: NOTIFICATION_AGREEMENT,
    notificationAgreement: agree,
  };
}

// -------------- TEXT --------------
function changeDataDisagreePeriod(period) {
  return {
    type: DATA_DISAGREE_PERIOD,
    dataDisagreePeriod: period,
  };
}
function changeLightDisagreePeriod(period) {
  return {
    type: LIGHT_DISAGREE_PERIOD,
    lightDisagreePeriod: period,
  };
}
function changeNotificationDisagreePeriod(period) {
  return {
    type: NOTIFICATION_DISAGREE_PERDIOD,
    notificationDisagreePeriod: period,
  };
}

export const ParameterActions = {
  changeDataAgreement,
  changeLightAgreement,
  changeNotificationAgreement,
  changeDataDisagreePeriod,
  changeLightDisagreePeriod,
  changeNotificationDisagreePeriod,
};

const ACTION_HANDLERS = {
  [DATA_AGREEMENT]: (state, action) => (
    { ...state, dataAgreement: action.dataAgreement }
  ),
  [LIGHT_AGREEMENT]: (state, action) => (
    { ...state, lightAgreement: action.lightAgreement }
  ),
  [NOTIFICATION_AGREEMENT]: (state, action) => (
    { ...state, notificationAgreement: action.notificationAgreement }
  ),
  [DATA_DISAGREE_PERIOD]: (state, action) => (
    { ...state, dataDisagreePeriod: action.dataDisagreePeriod }
  ),
  [LIGHT_DISAGREE_PERIOD]: (state, action) => (
    { ...state, lightDisagreePeriod: action.lightDisagreePeriod }
  ),
  [NOTIFICATION_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, notificationDisagreePeriod: action.notificationDisagreePeriod }
  ),
};

export const initParameter = {
  dataAgreement: true,
  lightAgreement: true,
  notificationAgreement: true,
  dataDisagreePeriod: null,
  lightDisagreePeriod: null,
  notificationDisagreePeriod: null,
};

export default function applicationReducer(state = initParameter, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
