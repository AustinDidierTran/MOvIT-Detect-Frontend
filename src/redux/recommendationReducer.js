/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

export const REDUCE_WEIGHT = 'REDUCE_WEIGHT';
export const REDUCE_SWELLING = 'REDUCE_SWELLING';
export const REDUCE_SLIDING_MOVING = 'REDUCE_SLIDING_MOVING';
export const REDUCE_SLIDING_REST = 'REDUCE_SLIDING_REST';
export const REDUCE_PAIN = 'REDUCE_PAIN';
export const ALLOW_REST = 'ALLOW_REST';
export const EASE_TRANSFERS = 'EASE_TRANSFERS';
export const IMPROVE_COMFORT = 'IMPROVE_COMFORT';
export const OTHER = 'OTHER';
export const TILT_FREQUENCY_WEIGHT = 'TILT_FREQUENCY_WEIGHT';
export const TILT_FREQUENCY_SWELLING = 'TILT_FREQUENCY_SWELLING';
export const TILT_LENGTH_WEIGHT = 'TILT_LENGTH_WEIGHT';
export const TILT_LENGTH_SWELLING = 'TILT_LENGTH_SWELLING';
export const TILT_ANGLE_WEIGHT = 'TILT_ANGLE_WEIGHT';
export const TILT_ANGLE_SWELLING = 'TILT_ANGLE_SWELLING';
export const TILT_ANGLE_MOVING = 'TILT_ANGLE_MOVING';
export const TILT_ANGLE_REST = 'TILT_ANGLE_REST';
export const REDUCE_PAIN_RECOMMENDATION = 'REDUCE_PAIN_RECOMMENDATION';
export const REDUCE_SWELLING_RECOMMENDATION = 'REDUCE_SWELLING_RECOMMENDATION';
export const ALLOW_REST_RECOMMENDATION = 'ALLOW_REST_RECOMMENDATION';
export const EASE_TRANSFERS_RECOMMENDATION = 'EASE_TRANSFERS_RECOMMENDATION';
export const IMPROVE_COMFORT_RECOMMENDATION = 'IMPROVE_COMFORT_RECOMMENDATION';
export const OTHER_RECOMMENDATION = 'OTHER_RECOMMENDATION';
export const OTHER_RECOMMENDATION_TITLE = 'OTHER_RECOMMENDATION_TITLE';

// -------------- CHECKBOX --------------
function changeReduceWeight() {
  return {
    type: REDUCE_WEIGHT,
  };
}
function changeReduceSwelling() {
  return {
    type: REDUCE_SWELLING,
  };
}
function changeReduceSlidingMoving() {
  return {
    type: REDUCE_SLIDING_MOVING,
  };
}
function changeReduceSlidingRest() {
  return {
    type: REDUCE_SLIDING_REST,
  };
}
function changeReducePain() {
  return {
    type: REDUCE_PAIN,
  };
}
function changeAllowRest() {
  return {
    type: ALLOW_REST,
  };
}
function changeEaseTransfers() {
  return {
    type: EASE_TRANSFERS,
  };
}
function changeImproveComfort() {
  return {
    type: IMPROVE_COMFORT,
  };
}
function changeOther() {
  return {
    type: OTHER,
  };
}

// -------------- SLIDERS --------------

function changeTiltFrequencyWeight(tiltFreq) {
  return {
    type: TILT_FREQUENCY_WEIGHT,
    tiltFrequencyWeight: tiltFreq,
  };
}
function changeTiltLengthWeight(tiltLen) {
  return {
    type: TILT_LENGTH_WEIGHT,
    tiltLengthWeight: tiltLen,
  };
}
function changeTiltAngleWeight(tiltAng) {
  return {
    type: TILT_ANGLE_WEIGHT,
    tiltAngleWeight: tiltAng,
  };
}
function changeTiltAngleMoving(tiltAngMoving) {
  return {
    type: TILT_ANGLE_MOVING,
    tiltAngleMoving: tiltAngMoving,
  };
}
function changeTiltAngleRest(tiltAngRest) {
  return {
    type: TILT_ANGLE_REST,
    tiltAngleRest: tiltAngRest,
  };
}

// -------------- TEXT --------------
function reducePainRecommendation(painRec) {
  return {
    type: REDUCE_PAIN_RECOMMENDATION,
    painRecommendation: painRec,
  };
}
function reduceSwellingRecommendation(swellingRec) {
  return {
    type: REDUCE_SWELLING_RECOMMENDATION,
    swellingRecommendation: swellingRec,
  };
}
function allowRestRecommendation(restRec) {
  return {
    type: ALLOW_REST_RECOMMENDATION,
    restRecommendation: restRec,
  };
}
function easeTransfersRecommendation(transferRec) {
  return {
    type: EASE_TRANSFERS_RECOMMENDATION,
    transferRecommendation: transferRec,
  };
}
function improveComfortRecommendation(comfortRec) {
  return {
    type: IMPROVE_COMFORT_RECOMMENDATION,
    comfortRecommendation: comfortRec,
  };
}
function otherRecommendation(otherRec) {
  return {
    type: OTHER_RECOMMENDATION,
    otherRecommendations: otherRec,
  };
}
function otherRecommendationTitle(otherRec) {
  return {
    type: OTHER_RECOMMENDATION_TITLE,
    otherRecommendationsTitle: otherRec,
  };
}
export const RecommendationActions = {
  changeReduceWeight,
  changeReduceSwelling,
  changeReduceSlidingMoving,
  changeReduceSlidingRest,
  changeReducePain,
  changeAllowRest,
  changeEaseTransfers,
  changeImproveComfort,
  changeOther,
  changeTiltFrequencyWeight,
  changeTiltLengthWeight,
  changeTiltAngleWeight,
  changeTiltAngleMoving,
  changeTiltAngleRest,
  reducePainRecommendation,
  reduceSwellingRecommendation,
  allowRestRecommendation,
  easeTransfersRecommendation,
  improveComfortRecommendation,
  otherRecommendation,
  otherRecommendationTitle,
};
const ACTION_HANDLERS = {
  [REDUCE_WEIGHT]: state => (
    { ...state, reduceWeight: !state.reduceWeight }
  ),
  [REDUCE_SWELLING]: state => (
    { ...state, reduceSwelling: !state.reduceSwelling }
  ),
  [REDUCE_SLIDING_MOVING]: state => (
    { ...state, reduceSlidingMoving: !state.reduceSlidingMoving }
  ),
  [REDUCE_SLIDING_REST]: state => (
    { ...state, reduceSlidingRest: !state.reduceSlidingRest }
  ),
  [REDUCE_PAIN]: state => (
    { ...state, reducePain: !state.reducePain }
  ),
  [ALLOW_REST]: state => (
    { ...state, allowRest: !state.allowRest }
  ),
  [EASE_TRANSFERS]: state => (
    { ...state, easeTransfers: !state.easeTransfers }
  ),
  [IMPROVE_COMFORT]: state => (
    { ...state, improveComfort: !state.improveComfort }
  ),
  [OTHER]: state => (
    { ...state, other: !state.other }
  ),
  [TILT_FREQUENCY_WEIGHT]: (state, action) => (
    { ...state, tiltFrequencyWeight: action.tiltFrequencyWeight }
  ),
  [TILT_LENGTH_WEIGHT]: (state, action) => (
    { ...state, tiltLengthWeight: action.tiltLengthWeight }
  ),
  [TILT_ANGLE_WEIGHT]: (state, action) => (
    { ...state, tiltAngleWeight: action.tiltAngleWeight }
  ),
  [TILT_ANGLE_MOVING]: (state, action) => (
    { ...state, tiltAngleMoving: action.tiltAngleMoving }
  ),
  [TILT_ANGLE_REST]: (state, action) => (
    { ...state, tiltAngleRest: action.tiltAngleRest }
  ),
  [REDUCE_PAIN_RECOMMENDATION]: (state, action) => (
    { ...state, painRecommendation: action.painRecommendation }
  ),
  [REDUCE_SWELLING_RECOMMENDATION]: (state, action) => (
    { ...state, swellingRecommendation: action.swellingRecommendation }
  ),
  [ALLOW_REST_RECOMMENDATION]: (state, action) => (
    { ...state, restRecommendation: action.restRecommendation }
  ),
  [EASE_TRANSFERS_RECOMMENDATION]: (state, action) => (
    { ...state, transferRecommendation: action.transferRecommendation }
  ),
  [IMPROVE_COMFORT_RECOMMENDATION]: (state, action) => (
    { ...state, comfortRecommendation: action.comfortRecommendation }
  ),
  [OTHER_RECOMMENDATION]: (state, action) => (
    { ...state, otherRecommendations: action.otherRecommendations }
  ),
  [OTHER_RECOMMENDATION_TITLE]: (state, action) => (
    { ...state, otherRecommendationsTitle: action.otherRecommendationsTitle }
  ),
};
export const initRecommendation = {
  reduceWeight: false,
  reduceSwelling: false,
  reduceSlidingMoving: false,
  reduceSlidingRest: false,
  reducePain: false,
  allowRest: false,
  easeTransfers: false,
  improveComfort: false,
  other: false,
  tiltFrequencyWeight: 0,
  tiltLengthWeight: 0,
  tiltAngleWeight: 0,
  tiltAngleMoving: 0,
  tiltAngleRest: 0,
  tiltFrequencyGoal: 0,
  tiltLengthGoal: 0,
  tiltAngleGoal: 0,
};
export default function applicationReducer(state = initRecommendation, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
