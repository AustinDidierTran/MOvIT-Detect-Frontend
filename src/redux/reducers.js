/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

import { combineReducers } from 'redux';
import applicationReducer from './applicationReducer';
import recommendationReducer from './recommendationReducer';
import goalReducer from './goalReducer';
import parameterReducer from './parameterReducer';
import configurationReducer from './configurationReducer';

export default combineReducers({
  applicationReducer,
  recommendationReducer,
  goalReducer,
  parameterReducer,
  configurationReducer,
});
