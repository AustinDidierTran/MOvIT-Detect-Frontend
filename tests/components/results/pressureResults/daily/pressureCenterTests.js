/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import PropTypes from 'prop-types';
import React from 'react';
import { Slider } from 'primereact/components/slider/Slider';
import axios from 'axios';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import PressureCenter from '../../../../../src/components/results/pressureResults/daily/pressureCenter';
import CustomCard from '../../../../../src/components/shared/card';
import { OFFSET, URL } from '../../../../../src/redux/applicationReducer';

Enzyme.configure({ adapter: new Adapter() });

function setFakeState(wrapper) {
  wrapper.setState({
    time: 10800000,
    hasErrors: false,
    isLoaded: true,
    index: 0,
    centers: [
      { x: 0.2, y: 0.3 },
      { x: 0.1, y: 0.2 },
      { x: 0.5, y: 0.7 },
    ],
    quadrants: [
      [
        { 0: { x: -1.2, y: 3.2 } },
        { 1: { x: 2.2, y: 3 } },
        { 2: { x: 3.2, y: -5.2 } },
        { 3: { x: -1.2, y: -3.2 } },
      ],
      [
        { 0: { x: -1.2, y: 3.2 } },
        { 1: { x: 0.2, y: -3 } },
        { 2: { x: 1.2, y: -2.2 } },
        { 3: { x: -2.2, y: -3.2 } },
      ],
      [
        { 0: { x: -3.2, y: 1.2 } },
        { 1: { x: 1.1, y: 2 } },
        { 2: { x: 3.1, y: -5.2 } },
        { 3: { x: -2.2, y: -5.2 } },
      ],
    ],
    hours: [10800000, 10800001, 10800002],
    currentCenter: { x: 0, y: 0 },
    currentQuadrants: [
      { x: 0, y: 0 },
      { x: 0, y: 0 },
      { x: 0, y: 0 },
      { x: 0, y: 0 },
    ],
  });
}

describe('PressureCenter Tests', () => {
  let wrapper;

  const initialState = { applicationReducer: { header: '' } };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    date: '1517720400000',
  };

  beforeEach(() => {
    wrapper = shallow(<PressureCenter store={store} {...props} />).dive();
  });

  it('should have proptypes', () => {
    const actualValue = PressureCenter.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      date: PropTypes.instanceOf(Date),
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should initialize the states', () => {
    expect(wrapper.state('isLoaded')).toEqual(false);
    expect(wrapper.state('hasErrors')).toEqual(false);
    expect(wrapper.state('time')).toEqual(0);
    expect(wrapper.state('index')).toEqual(0);
    expect(wrapper.state('currentCenter')).toEqual({ x: 0, y: 0 });
    expect(wrapper.state('currentQuadrants')).toEqual([
      { x: 0, y: 0 },
      { x: 0, y: 0 },
      { x: 0, y: 0 },
      { x: 0, y: 0 },
    ]);
    expect(wrapper.state('quadrants')).toEqual([]);
    expect(wrapper.state('centers')).toEqual([]);
    expect(wrapper.state('hours')).toEqual([]);
  });

  it('should get pressure data', async () => {
    const mock = new MockAdapter(axios);
    const data = {
      response: true,
    };

    mock.onGet(`${URL}gravityCenter?Day=1517720400000,offset=${OFFSET}`).reply(200, data);

    const response = await wrapper.instance().getPressureData(props.date);

    expect(wrapper.state('hasErrors')).toEqual(false);
    expect(response).toEqual(data);
  });

  it('should set the index', () => {
    setFakeState(wrapper);

    wrapper.instance().setIndex(1);

    expect(wrapper.state('index')).toEqual(1);
    expect(wrapper.state('currentCenter')).toEqual({ x: 0.1, y: 0.2 });
    expect(wrapper.state('currentQuadrants')).toEqual([
      { 0: { x: -1.2, y: 3.2 } },
      { 1: { x: 0.2, y: -3 } },
      { 2: { x: 1.2, y: -2.2 } },
      { 3: { x: -2.2, y: -3.2 } },
    ]);
    expect(wrapper.state('time')).toEqual(10800001);
  });

  it('should load the pressure data', () => {
    const data = {
      10800000: {
        center: {
          x: -0.2,
          y: 0.3,
        },
        quadrants: [
          { 0: { x: -1.2, y: 3.2 } },
          { 1: { x: 2.2, y: 3 } },
          { 2: { x: 3.2, y: -5.2 } },
          { 3: { x: -1.2, y: -3.2 } },
        ],
      },
    };
    const spy = sinon.spy(wrapper.instance(), 'setIndex');

    wrapper.instance().loadPressureData(data);

    expect(spy.calledOnce).toEqual(true);
    expect(spy.getCalls()[0].args[0]).toEqual(0);
    expect(wrapper.state('isLoaded')).toEqual(true);
    expect(wrapper.state('hours')).toEqual(['10800000']);
    expect(wrapper.state('centers')).toEqual([{ x: -0.2, y: 0.3 }]);
    expect(wrapper.state('quadrants')).toEqual([[
      { 0: { x: -1.2, y: 3.2 } },
      { 1: { x: 2.2, y: 3 } },
      { 2: { x: 3.2, y: -5.2 } },
      { 3: { x: -1.2, y: -3.2 } },
    ]]);
  });

  it('should trigger setIndex when simulating a change event on the Slider', () => {
    setFakeState(wrapper);

    const spy = sinon.spy(wrapper.instance(), 'setIndex');

    wrapper.find(CustomCard).dive().find(Slider).simulate('change', { value: 1 });

    expect(spy.calledOnce).toEqual(true);
    expect(spy.getCalls()[0].args[0]).toEqual(1);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
