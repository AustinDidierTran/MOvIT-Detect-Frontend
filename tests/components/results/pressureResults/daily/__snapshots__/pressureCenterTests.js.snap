// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressureCenter Tests should match the snapshot 1`] = `
<div
  className="container"
  id="dailyPressureCenter"
  style={
    Object {
      "center": Object {
        "textAlign": "center",
      },
      "height": "60%",
      "marginBottom": "10%",
      "width": "100%",
    }
  }
>
  <CustomCard
    element={<Loading />}
    header={<h4 />}
  />
</div>
`;
